// allure-js-commons/dist/src/AllureTest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllureTest = void 0;
const ExecutableItemWrapper_1 = require("./ExecutableItemWrapper");
const constructors_1 = require("./constructors");
class AllureTest extends ExecutableItemWrapper_1.ExecutableItemWrapper {
    constructor(runtime, start = Date.now()) {
        super(constructors_1.testResult());
        this.runtime = runtime;
        this.testResult = this.wrappedItem;
        this.testResult.start = start;
    }
    endTest(stop = Date.now()) {
        this.testResult.stop = stop;
        this.runtime.writeResult(this.testResult);
    }
    get uuid() {
        return this.testResult.uuid;
    }
    set historyId(id) {
        this.testResult.historyId = id;
    }
    set fullName(fullName) {
        this.testResult.fullName = fullName;
    }
    addLabel(name, value) {
        this.testResult.labels.push({ name, value });
    }
    addLink(url, name, type) {
        this.testResult.links.push({ name, url, type });
    }
}
exports.AllureTest = AllureTest;
